### PR TITLE
TypeORM shim to use entities in Next.js frontend

### PR DESCRIPTION
**Pull request by jaredcwhite**
(Coming soon with DB backend, not yet activated)

This was part of a previous PR that got overwritten, but since we haven't yet fully switched to the DB backend, we wouldn't have wanted to roll it out without keeping it under wraps for the time being. So this PR adds the functionality for the frontend to potentially use the backend TS interfaces but doesn't yet activate it.

### CODE REVIEW

File: apps/public-reference/src/typeorm-model-shim.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
+/* eslint-disable */
+
+// columns
```
Comment (bencpeters) [inline apps/public-reference/src/typeorm-model-shim.js:3]: I assume this file is essentially cut & pasted from [the shim provided by TypeORM](https://github.com/typeorm/typeorm/blob/master/extra/typeorm-model-shim.js). Do we want to host our own version of this, or should we literally use the shim provided by TypeORM? We could adjust the webpack config to look in `node_modules`:

```javascript
resolve: { // see: https://webpack.js.org/configuration/resolve/
  alias: {
    typeorm: path.resolve(__dirname, "../node_modules/typeorm/typeorm-model-shim")
  }
}
```

Comment (jaredcwhite) [reply apps/public-reference/src/typeorm-model-shim.js:3]: I had to make one modification, but it was something they actually should have included in that file, so theoretically if it got upstreamed we could pull it from the module dir. But for now we need to keep it in-repo.

Comment (bencpeters) [reply apps/public-reference/src/typeorm-model-shim.js:3]: @jaredcwhite Does [this PR](https://github.com/typeorm/typeorm/pull/6354) resolve the issue?

Comment (jaredcwhite) [reply apps/public-reference/src/typeorm-model-shim.js:3]: @bencpeters Unfortunately not — the shim is missing `BaseEntity` for some reason, I had to add it in.

**Comment by bk3c** [comment]
Closing this in favor of swagger-axios-codegen and/or its successor.

**Pull request closed without merging**